e2e_tests/tests/experiment/experiment.py

```diff
@@ -664,11 +664,12 @@ def run_failure_test_with_temp_config(
     config: Dict[Any, Any],
     model_def_path: str,
     error_str: Optional[str] = None,
-) -> None:
+) -> int:
     with tempfile.NamedTemporaryFile() as tf:
         with open(tf.name, "w") as f:
             yaml.dump(config, f)
-        run_failure_test(tf.name, model_def_path, error_str=error_str)
+        experiment_id = run_failure_test(tf.name, model_def_path, error_str=error_str)
```

Comment:
you can also just `return run_failure_test(...)`.  The context managers are still respected before returning control to the caller